pmd-java/src/main/java/net/sourceforge/pmd/lang/java/rule/errorprone/BeanMembersShouldSerializeRule.java

```diff
@@ -74,43 +64,35 @@ public Object visit(ASTClassOrInterfaceDeclaration node, Object data) {
         }
 
         if (JavaRuleUtil.hasLombokAnnotation(node)) {
-            return super.visit(node, data);
-        }
-
-        Map<MethodNameDeclaration, List<NameOccurrence>> methods = node.getScope().getEnclosingScope(ClassScope.class)
-                .getMethodDeclarations();
-        List<ASTMethodDeclaration> getSetMethList = new ArrayList<>(methods.size());
-        for (MethodNameDeclaration d : methods.keySet()) {
-            ASTMethodDeclaration mnd = d.getDeclarator();
-            if (isBeanAccessor(mnd)) {
-                getSetMethList.add(mnd);
-            }
+            return data;
         }
 
-        String[] methNameArray = imagesOf(getSetMethList);
-
-        Arrays.sort(methNameArray);
-
-        Map<VariableNameDeclaration, List<NameOccurrence>> vars = node.getScope()
-                .getDeclarations(VariableNameDeclaration.class);
-        for (Map.Entry<VariableNameDeclaration, List<NameOccurrence>> entry : vars.entrySet()) {
-            VariableNameDeclaration decl = entry.getKey();
-            AccessNode accessNodeParent = decl.getAccessNodeParent();
-            if (entry.getValue().isEmpty() || accessNodeParent.isTransient() || accessNodeParent.isStatic()
-                    || hasIgnoredAnnotation((Annotatable) accessNodeParent)) {
+        List<String> accessors = node.descendants(ASTMethodDeclaration.class)
+                .filter(JavaRuleUtil::isGetterOrSetter)
+                .toStream()
+                .map(m -> m.getName())
+                .collect(Collectors.toList());
+        Collections.sort(accessors);
+
+        List<@NonNull ASTVariableDeclaratorId> fields = node.descendants(ASTFieldDeclaration.class)
+                .descendants(ASTVariableDeclaratorId.class).toList();
+        for (ASTVariableDeclaratorId field : fields) {
+            if (field.getLocalUsages().isEmpty()
+                    || field.getModifiers().hasAny(JModifier.TRANSIENT, JModifier.STATIC)
+                    || hasIgnoredAnnotation(field)) {
                 continue;
             }
-            String varName = StringUtils.capitalize(trimIfPrefix(decl.getImage()));
-            boolean hasGetMethod = Arrays.binarySearch(methNameArray, "get" + varName) >= 0
-                    || Arrays.binarySearch(methNameArray, "is" + varName) >= 0;
-            boolean hasSetMethod = Arrays.binarySearch(methNameArray, "set" + varName) >= 0;
+            String varName = StringUtils.capitalize(trimIfPrefix(field.getName()));
+            boolean hasGetMethod = Collections.binarySearch(accessors, "get" + varName) >= 0
```

Comment:
Ok so AFAIU this method improvises a `Set<String>` with a sorted array/list. Let's clarify that